contrib/cryptomb/private_key_providers/source/cryptomb_private_key_provider.cc

```diff
@@ -0,0 +1,793 @@
+#include "contrib/cryptomb/private_key_providers/source/cryptomb_private_key_provider.h"
+
+#include <memory>
+
+#include "envoy/registry/registry.h"
+#include "envoy/server/transport_socket_config.h"
+
+#include "source/common/config/datasource.h"
+
+#include "openssl/ec.h"
+#include "openssl/ssl.h"
+
+// BoringSSL internal definitions, needed for calculating ECDSA ephemeral key.
+// TODO(ipuustin): ask BoringSSL maintainers to expose these in the BoringSSL headers?
+#define BN_BYTES 8
+#define EC_MAX_BYTES 66
+#define EC_MAX_WORDS ((EC_MAX_BYTES + BN_BYTES - 1) / BN_BYTES)
+
+// NOLINTNEXTLINE(modernize-use-using)
+typedef union {
+  uint8_t bytes[EC_MAX_BYTES];
+  BN_ULONG words[EC_MAX_WORDS];
+} EC_SCALAR;
+
+extern "C" {
+// NOLINTNEXTLINE(readability-identifier-naming)
+int ec_random_nonzero_scalar(const EC_GROUP* group, EC_SCALAR* out,
```

Comment:
I hadn't realized earlier that you're accessing an internal BoringSSL symbol here. This could break anytime we update the Envoy BoringSSL version. One option is that if this extension stops compiling, or tests stop passing, during a version update, we remove it from the contrib build until you can fix it. But I'm somewhat concerned about the increased friction of being able to handle core dependency updates from this.

cc @envoyproxy/senior-maintainers

Reply:
Yeah IMO we need to fix this before merging.